pkg/apis/pipeline/v1beta1/taskrun_validation_test.go

```diff
@@ -112,6 +133,59 @@ func TestTaskRun_Validate(t *testing.T) {
 			},
 		},
 		wc: config.EnableAlphaAPIFields,
+	}, {
+		name: "propagating object params with taskrun",
+		taskRun: &v1beta1.TaskRun{
+			ObjectMeta: metav1.ObjectMeta{Name: "tr"},
+			Spec: v1beta1.TaskRunSpec{
+				Params: []v1beta1.Param{{
+					Name: "task-words",
+					Value: v1beta1.ParamValue{
+						Type:      v1beta1.ParamTypeObject,
+						ObjectVal: map[string]string{"hello": "task run"},
+					},
+				}},
+				TaskSpec: &v1beta1.TaskSpec{
+					Steps: []v1beta1.Step{{
+						Name:    "echo",
+						Image:   "ubuntu",
+						Command: []string{"echo"},
+						Args:    []string{"$(params.task-words.hello)"},
+					}},
+				},
+			},
+		},
+		wc: config.EnableAlphaAPIFields,
+	}, {
+		name: "propagating object params with taskrun no value provided",
+		taskRun: &v1beta1.TaskRun{
+			ObjectMeta: metav1.ObjectMeta{Name: "tr"},
+			Spec: v1beta1.TaskRunSpec{
+				Params: []v1beta1.Param{{
+					Name: "task-words",
+					Value: v1beta1.ParamValue{
+						Type: v1beta1.ParamTypeObject,
+					},
+				}},
+				TaskSpec: &v1beta1.TaskSpec{
+					Params: []v1beta1.ParamSpec{{
+						Name: "task-words",
+						Type: v1beta1.ParamTypeObject,
+						Default: &v1beta1.ParamValue{
+							Type:      v1beta1.ParamTypeObject,
+							ObjectVal: map[string]string{"hello": "task run def"},
+						},
+					}},
+					Steps: []v1beta1.Step{{
+						Name:    "echo",
+						Image:   "ubuntu",
+						Command: []string{"echo"},
+						Args:    []string{"$(params.task-words.hello)"},
+					}},
+				},
+			},
+		},
+		wc: config.EnableAlphaAPIFields,
```

Comment:
I am a bit doubt if this a real case for propagating param. If no value is provided in taskrun, why does user write something about that param in taskrun?

Reply:
I had this in place to increase coverage. Because of the behaviour of propagated params, when we combine defaults with provided values, I need to cover different cases.